### PR TITLE
Add upgrade notes

### PR DESCRIPTION
**Pull request by crazy-max**
Ref. #137 #140 #141

CHANGELOG have also been removed and will be available through GitHub releases.

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,17 @@
 [![Test workflow](https://img.shields.io/github/workflow/status/docker/build-push-action/test?label=test&logo=github&style=flat-square)](https://github.com/docker/build-push-action/actions?workflow=test)
 [![Codecov](https://img.shields.io/codecov/c/github/docker/build-push-action?logo=codecov&style=flat-square)](https://codecov.io/gh/docker/build-push-action)
 
+## Upgrade
+
+`v2` of this action changes drastically and now uses Docker [Buildx](https://github.com/docker/buildx). It works with
```
Comment (tonistiigi) [inline README.md:9]: s/changes drastically/includes significant updates/

File: README.md [hunk 2]
```diff
@@ -4,6 +4,17 @@
...
+3 new actions ([login](https://github.com/docker/login-action), [setup-buildx](https://github.com/docker/setup-buildx-action)
```
Comment (tonistiigi) [inline README.md:10]: "new optional actions"

File: README.md [hunk 3]
```diff
@@ -4,6 +4,17 @@
 [![Test workflow](https://img.shields.io/github/workflow/status/docker/build-push-action/test?label=test&logo=github&style=flat-square)](https://github.com/docker/build-push-action/actions?workflow=test)
 [![Codecov](https://img.shields.io/codecov/c/github/docker/build-push-action?logo=codecov&style=flat-square)](https://codecov.io/gh/docker/build-push-action)
 
+## Upgrade
```
Comment (tonistiigi) [inline README.md:7]: "Upgrade from v1" ?

File: UPGRADE.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
+# Upgrade notes
+
+## v1 to v2
+
+* Rename `path` input to `context`
```
Comment (tonistiigi) [inline UPGRADE.md:5]: Input `path` is now called `context` for consistency with other Docker build tools. 

same for others

File: UPGRADE.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+* Rename `dockerfile` input to `file`
+* Rename `always_pull` input to `pull`
+* Add `builder` input to be able to choose a builder instance through our [setup-buildx action](https://github.com/docker/setup-buildx-action)
+* Add [`platforms`](https://github.com/docker/buildx#---platformvaluevalue) input
```
Comment (tonistiigi) [inline UPGRADE.md:9]: .. to support multi-platfrom builds

File: UPGRADE.md (hunk 4 above)
Comment (tonistiigi) [reply UPGRADE.md:5]: `path` defaults to current git repository so checkout action is not required in a workflow.

File: README.md (hunk 3 above)
Comment (crazy-max) [reply README.md]: LGTM

File: UPGRADE.md (hunk 4 above)
Comment (tonistiigi) [reply UPGRADE.md]: Same for `dockerfile` `always_pull`

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**